# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# Filename: package/.../rrdtool/rrdtool-1.2.19-rrdcgi.patch
# Copyright (C) 2007 The OpenSDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

Printstrftime on rrd_cgi.c returns an illegal value.
It causes segmentation fault error. (Tatsuki Makino tatsuki_makino hotmail.com)

--- ./src/rrd_cgi.c	(Revision 1000)
+++ ./src/rrd_cgi.c	(Revision 1001)
@@ -574,7 +574,7 @@
 	/* Make sure that we were given the right number of args */
 	if( argc != 4) {
 		rrd_set_error( "wrong number of args %d", argc);
-		return (char *) -1;
+		return stralloc("");
 	}
 
 	/* Init start and end time */
@@ -584,14 +584,14 @@
 	/* Parse the start and end times we were given */
 	if( (parsetime_error = parsetime( args[1], &start_tv))) {
 		rrd_set_error( "start time: %s", parsetime_error);
-		return (char *) -1;
+		return stralloc("");
 	}
 	if( (parsetime_error = parsetime( args[2], &end_tv))) {
 		rrd_set_error( "end time: %s", parsetime_error);
-		return (char *) -1;
+		return stralloc("");
 	}
 	if( proc_start_end( &start_tv, &end_tv, &start_tmp, &end_tmp) == -1) {
-		return (char *) -1;
+		return stralloc("");
 	}
 
 	/* Do we do the start or end */
@@ -603,7 +603,7 @@
 	}
 	else {
 		rrd_set_error( "start/end not found in '%s'", args[0]);
-		return (char *) -1;
+		return stralloc("");
 	}
 
 	/* now format it */
@@ -612,7 +612,7 @@
 	}
 	else {
 		rrd_set_error( "strftime failed");
-		return (char *) -1;
+		return stralloc("");
 	}
 }
 
